dlc-wallet/src/main/scala/org/bitcoins/dlc/wallet/internal/DLCTransactionProcessing.scala

```diff
@@ -91,10 +95,30 @@ private[bitcoins] trait DLCTransactionProcessing extends TransactionProcessing {
         (announcements, announcementData, nonceDbs) <- getDLCAnnouncementDbs(
           dlcDb.dlcId)
 
-        cet <-
-          transactionDAO
+        cet <- for {
+          txFromDb <- transactionDAO
             .read(dlcDb.closingTxIdOpt.get)
-            .map(_.get.transaction.asInstanceOf[WitnessTransaction])
```

Comment:
if we aren't getting the witness version of the tx there might be a bug in the node where we aren't requesting the witness version of the tx